google/cloud/bigquery/_pandas_helpers.py

```diff
@@ -33,6 +33,18 @@
 except ImportError:  # pragma: NO COVER
     pyarrow = None
 
+try:
+    from google.cloud.bigquery_storage import ArrowSerializationOptions
+except ImportError:
+    _ARROW_COMPRESSION_SUPPORT = False
+else:
+    import pkg_resources
+
+    # Having BQ Storage available implies that pyarrow is available, too.
+    _ARROW_COMPRESSION_SUPPORT = pkg_resources.get_distribution(
+        "pyarrow"
+    ).parsed_version >= pkg_resources.parse_version("1.0.0")
```

Comment:
Do we actually need `pyarrow` version check?

The `bqstorage` extra already pins the minimum `pyarrow` version to `1.0.0`, thus I suppose if somebody somehow installs a less recent version, it can be considered an error on the user side?

Reply:
Sounds good to me (removing this check). Since all of our extras require `> 1.0.0`, I'm okay with failing.

Counterpoint though is: what error will they get when `pyarrow` is too low? I wonder if we should file an FR to check for minimum versions and throw nicer errors? It might help with some of the issues like https://github.com/googleapis/python-bigquery/issues/556. This would mean keeping minimum versions in sync with 3 locations: `setup.py`, `constraints.txt`, and `version_check.py`, though.

Reply:
The new pip dependency resolver ([announcement](https://discuss.python.org/t/announcement-pip-20-2-release/4863)) is significantly stricter, thus I suppose this will become a no-issue once its use is widespread enough and/or becomes a default. It shouldn't even be possible to install an incompatible `pyarrow` version with it.

Yes, we can open a FR to discuss there if covering this presumably corner-case scenario in the transitional period is worth doing, considering the problem with keeping the min versions in sync.